app.py

```diff
@@ -102,357 +125,270 @@ def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
     prevent_initial_call=True,
 )
 def run_optimization_cqm(
-    run_click: int, model: str, solver: str, scenario: str, time_limit: int
-) -> tuple[go.Figure, go.Figure, str, dict]:
+    run_click: int, model: int, solvers: list[int], scenario: str, time_limit: int
+) -> tuple[go.Figure, go.Figure, str, str, bool, bool]:
     """Runs optimization using the D-Wave hybrid solver.
 
     Args:
         run_click (int): The number of times the run button has been
             clicked.
-        model (str): The model to use for the optimization.
-        solver (str): The solver to use for the optimization.
+        model (int): The model to use for the optimization.
+        solvers (list[int]): The solvers that have been selected.
         scenario (str): The scenario to use for the optimization.
         time_limit (int): The time limit for the optimization.
 
     Returns:
-        go.Figure: Gantt chart for the D-Wave hybrid solver
-        go.Figure: Results table for the D-Wave hybrid solver
-        str: Class name for the D-Wave tab
-        dict: Style for the D-Wave summary table
+        go.Figure: Gantt chart for the D-Wave hybrid solver.
+        go.Figure: Results table for the D-Wave hybrid solver.
+        str: Class name for the D-Wave tab.
+        str: The label for the D-Wave tab.
+        bool: True if D-Wave tab should be disabled, False otherwise.
+        bool: Whether D-Wave solver is running.
     """
-    if run_click == 0 or ctx.triggered_id != "run-button":
-        empty_figure = get_empty_figure("Run optimization to see results")
-        return empty_figure, "tab"
-    if ctx.triggered_id == "run-button":
-        if "Hybrid" in solver:
-            model_data = JobShopData()
-            filename = SCENARIOS[scenario]
-            model_data.load_from_file(DATA_PATH.joinpath(filename), resource_names=RESOURCE_NAMES)
-            allow_quadratic_constraints = model == "QM"
-            start = time.time()
-            results = run_shop_scheduler(
-                model_data,
-                use_mip_solver=False,
-                allow_quadratic_constraints=allow_quadratic_constraints,
-                solver_time_limit=time_limit,
-            )
-            end = time.time()
-            fig = generate_gantt_chart(df=results, y_axis="Job", color="Resource")
-            table = generate_output_table(results["Finish"].max(), time_limit, int(end - start))
-            return fig, table, "tab-success", {"visibility": "visible"}
-        else:
-            time.sleep(0.1)
-            empty_figure = get_empty_figure(
-                HTML_CONFIGS["solver_messages"]["dwave"]["solver_not_chosen"]
-            )
-            table = generate_output_table(0, 0, 0)
-            return empty_figure, table, "tab-warning", {"visibility": "hidden"}
-    else:
+    if ctx.triggered_id != "run-button" or run_click == 0:
         raise PreventUpdate
 
+    if not SamplerType.HYBRID.value in solvers:
```

Comment:
Minor, but preferred.
```suggestion
    if SamplerType.HYBRID.value not in solvers:
```